src/utils/polygon.cpp

```diff
@@ -295,6 +295,34 @@ Polygons ConstPolygonRef::offset(int distance, ClipperLib::JoinType join_type, d
     return ret;
 }
 
+bool lineLineIntersection(Point A, Point B, Point C, Point D, Point* output)
```

Comment:
We should probably make this part of a namespace class like `LinearAlg2D`. And use lowercase variable names.

Reply:
Use a reference for the output, rather than a pointer. You don't need the reference to change to a different address.